File: ruby/count_ip_addresses.rb
def ipsBetween(start, ending)
  ip_address_to_integer(ending) - ip_address_to_integer(start)
end

def ip_address_to_integer(ip_address)
  numbers = ip_address.split('.').map(&:to_i)
  numbers[0] * (256**3) + numbers[1] * (256**2) + numbers[2] * (256**1) + numbers[3] * (256**0)
end

File: python/count_ip_addresses.py
def ips_between(start, ending):
    return ip_address_to_integer(ending) - ip_address_to_integer(start)

def ip_address_to_integer(ip_address):
    numbers = list(map(int, ip_address.split('.')))
    return numbers[0] * (256**3) + numbers[1] * (256**2) + numbers[2] * (256**1) + numbers[3] * (256**0)